src/App/Pages/Accounts/LoginPasswordlessViewModel.cs

```diff
@@ -0,0 +1,191 @@
+        {
+            get => _logInAttempByLabel;
+            set => SetProperty(ref _logInAttempByLabel, value);
+        }
+
+        public string DeviceType
+        {
+            get => _deviceType;
+            set => SetProperty(ref _deviceType, value);
+        }
+
+        public string IpAddress
+        {
+            get => _ipAddress;
+            set => SetProperty(ref _ipAddress, value);
+        }
+
+        public string NearLocation
+        {
+            get => _nearLocation;
+            set => SetProperty(ref _nearLocation, value);
+        }
+
+        public DateTime RequestDate
+        {
+            get => _requestDate;
+            set
+            {
+                TimeOfRequestText = CreateRequestDate();
+                SetProperty(ref _requestDate, value);
+            }
+        }
+
+        public string TimeOfRequestText
+        {
+            get => _timeOfRequest;
+            set
+            {
+                SetProperty(ref _timeOfRequest, value);
+            }
+        }
+
+        private FormattedString CreateFingerprintPhrase(string fingerprintPhrase)
+        {
+            var fingerprintList = fingerprintPhrase.Split('-').ToList();
+            var fs = new FormattedString();
+            var lastFingerprint = fingerprintList.LastOrDefault();
+
+            foreach (var fingerprint in fingerprintList)
+            {
+                fs.Spans.Add(new Span
+                {
+                    Text = fingerprint
+                });
+
+                if (fingerprint == lastFingerprint)
+                {
+                    break;
+                }
+
+                fs.Spans.Add(new Span
+                {
+                    Text = "-",
+                    TextColor = ThemeManager.GetResourceColor("DangerColor")
+                });
+            }
+
+            return fs;
+        }
+
+        private string CreateRequestDate()
+        {
+            var minutesSinceRequest = RequestDate.ToUniversalTime().Minute - DateTime.UtcNow.Minute;
+            if (minutesSinceRequest < 5)
+            {
+                return AppResources.JustNow;
+            }
+            if (minutesSinceRequest < 59)
+            {
+                return $"{minutesSinceRequest} {AppResources.MinutesAgo}";
+            }
+
+            return RequestDate.ToShortTimeString();
+        }
+
+        private async Task AcceptRequestAsync()
+        {
+            try
+            {
+                var res = await _authService.LogInPasswordlessAcceptAsync();
+                await Page.Navigation.PopModalAsync();
+                _platformUtilsService.ShowToast("info", null, AppResources.LogInAccepted);
+            }
+            catch (Exception ex)
+            {
+                _logger.Exception(ex);
+            }
+        }
+
+        private async Task RejectRequestAsync()
+        {
+            try
+            {
+                var res = await _authService.LogInPasswordlessRejectAsync();
```

Comment:
I believe this will have an API call, so you should display a loading here. Same in `AcceptRequestAsync`